### DISCUSSION TITLE
Slow notebook loading due to pypi.org request - Add configuration to use PyPI mirror

### DISCUSSION THREAD

**Discussion by JackYinpei** [jupyterlite/jupyterlite#1254] [Q&A] [answered]
![image](https://github.com/jupyterlite/jupyterlite/assets/49555788/66eb30ee-4e4c-4b30-8c3b-9e41ad37d836)

Hello, 

I encountered a problem when loading the ipynb interface of the notebook. 

During the loading process, it requests the package "https://pypi.org/simple/comm", but accessing servers abroad is very slow in our country, (as shown in the image, it took over 20 seconds to finish loading 7kb of data), and it is highly likely to fail, (if requesting this package fails, the interface cannot continue running afterwards). 

Therefore, I want to use a mirror address to replace `pypi.org`, but I did not find any configuration options to let me set or use `micropip.set_index_urls`. 

May I ask if there are any configuration items I'm not aware of that can help me achieve this functionality? 

If you have any suggestions or advice, I would be most grateful. Thanks in advance.

**Comment by jtpio** [comment, marked as answer]
> May I ask if there are any configuration items I'm not aware of that can help me achieve this functionality?

If you're using the Pyodide kernel maybe having a look at the following configuration can help: https://jupyterlite.readthedocs.io/en/stable/howto/pyodide/wheels.html

Otherwise you might want to have a look at the `jupyterlite-xeus-python` kernel to pre-install packages and avoid requests to PyPI: https://jupyterlite.readthedocs.io/en/stable/howto/xeus-python/preinstalled_packages.html
